src/widgets/index/index.ts

```diff
@@ -569,3 +673,21 @@ const index = (props: IndexProps): Index => {
 };
 
 export default index;
+
+function storeRenderState({
```

Comment:
I think it would be better to have it as a method in `index`, it would simplify the signature (no need to pass `instantSearchInstance` and `parent`).

Reply:
We call it only twice (and will call it only one time once when remove the `init` method). Besides, it would make it available on the `index` widget scope, which we want to avoid.

Reply:
If you do not want to expose it you can set it as a private function, like `createURL`.

Reply:
Right, but we still need to pass the `instantSearchInstance` reference from `init` and `render` to mutate it, so it won't behave the same if we rely on `localInstantSearchInstance`.

Reply:
Still winning one parameter :smile:  , and it is more coherent with what we are already doing in this module.

Reply:
I agree with François, we want to avoid exposing a public API we didn't think about, private isn't really private